src/main.js

```diff
@@ -93,7 +94,16 @@ module.exports = {
       }
     }))
 
-    this.subscriptions.add(atom.commands.add('atom-text-editor', {
+    // Marks each JavaScript editor with a CSS class so that
+    // we can enable commands only for JavaScript editors.
+    this.subscriptions.add(atom.workspace.observeTextEditors((textEditor) => {
+      if (textEditor.getRootScopeDescriptor().scopes
```

Comment:
Hmmmm, I think I would say that it will work for providers with well defined allowed scope lists (like TSLint), and in fact for those the `getRootScopeDescriptor` might even be a better choice.

For providers like this one though where the allowed scopes can change at any time, this doesn't really make much sense, and to make it work at all would involve watching quite a bit of the components of Atom to ensure you get everything, so unfortunately I think for `linter-eslint` at least this doesn't really make much sense.